File: Python/JSON Payload/merged-pdf.py
import requests
import json

with open('/path/to/first_file', 'rb') as f:
    upload_first_file_data = f.read()

print("Uploading first PDF file...")
upload_first_file_response = requests.post(url='https://api.pdfrest.com/upload',
                    data=upload_first_file_data,
                    headers={'Content-Type': 'application/octet-stream', 'content-filename': 'first_file.pdf', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})

print("First upload response status code: " + str(upload_first_file_response.status_code))


with open('/path/to/second_file', 'rb') as f:
    upload_second_file_data  = f.read()

print("Uploading second PDF file...")
upload_second_file_response = requests.post(url='https://api.pdfrest.com/upload',
                    data=upload_second_file_data ,
                    headers={'Content-Type': 'application/octet-stream', 'content-filename': 'second_file.pdf', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})

print("Second upload response status code: " + str(upload_second_file_response.status_code))

if upload_first_file_response.ok and upload_second_file_response.ok:
    upload_first_file_response_json = upload_first_file_response.json()
    print(json.dumps(upload_first_file_response_json, indent = 2))

    upload_second_file_response_json = upload_second_file_response.json()
    print(json.dumps(upload_second_file_response_json, indent = 2))


    uploaded_first_file_id = upload_first_file_response_json['files'][0]['id']
    uploaded_second_file_id = upload_second_file_response_json['files'][0]['id']
    merge_data = { "id" : [uploaded_first_file_id, uploaded_second_file_id], "pages":[1,1], "type":["id", "id"] }
    print(json.dumps(merge_data, indent = 2))


    print("Processing file...")
    merge_response = requests.post(url='https://api.pdfrest.com/merged-pdf',
                        data=json.dumps(merge_data),
                        headers={'Content-Type': 'application/json', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})



    print("Processing response status code: " + str(merge_response.status_code))
    if merge_response.ok:
        merge_response_json = merge_response.json()
        print(json.dumps(merge_response_json, indent = 2))

    else:
        print(merge_response.text)
else:
    print(upload_first_file_response.text)
    print(upload_second_file_response.text)

File: DotNET/Endpoint Examples/JSON Payload/merged-pdf.cs

/*
 * What this sample does:
 * - Called from Program.cs to upload two files, then request merged-pdf.
 *
 * Setup (environment):
 * - Copy .env.example to .env
 * - Set PDFREST_API_KEY=your_api_key_here
 * - Optional: set PDFREST_URL to override the API region. For EU/GDPR compliance and proximity, use:
 *     PDFREST_URL=https://eu-api.pdfrest.com
 *   For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work
 *
 * Usage:
 *   dotnet run -- merged-pdf /path/to/file1.pdf /path/to/file2.pdf
 *
 * Output:
 * - Prints JSON responses; non-2xx results exit non-zero.
 */
using Newtonsoft.Json.Linq;
using System.Text;

namespace Samples.EndpointExamples.JsonPayload
{
    public static class MergedPdf
    {
        public static async Task Execute(string[] args)
        {
            if (args == null || args.Length < 2)
            {
                Console.Error.WriteLine("merged-pdf requires <firstFile> <secondFile>");
                Environment.Exit(1);
                return;
            }

            var firstPath = args[0];
            var secondPath = args[1];
            if (!File.Exists(firstPath) || !File.Exists(secondPath))
            {
                Console.Error.WriteLine("One or more files not found.");
                Environment.Exit(1);
                return;
            }

            var apiKey = Environment.GetEnvironmentVariable("PDFREST_API_KEY");
            if (string.IsNullOrWhiteSpace(apiKey))
            {
                Console.Error.WriteLine("Missing required environment variable: PDFREST_API_KEY");
                Environment.Exit(1);
                return;
            }
            var baseUrl = Environment.GetEnvironmentVariable("PDFREST_URL") ?? "https://api.pdfrest.com";

            using (var httpClient = new HttpClient { BaseAddress = new Uri(baseUrl) })
            {
                using (var firstUploadRequest = new HttpRequestMessage(HttpMethod.Post, "upload"))
                {
                    firstUploadRequest.Headers.TryAddWithoutValidation("Api-Key", apiKey);
                    firstUploadRequest.Headers.Accept.Add(new("application/json"));

                    var firstUploadByteArray = File.ReadAllBytes(firstPath);
                    var firstUploadByteAryContent = new ByteArrayContent(firstUploadByteArray);
                    firstUploadByteAryContent.Headers.TryAddWithoutValidation("Content-Type", "application/octet-stream");
                    firstUploadByteAryContent.Headers.TryAddWithoutValidation("Content-Filename", Path.GetFileName(firstPath));


                    firstUploadRequest.Content = firstUploadByteAryContent;
                    var firstUploadResponse = await httpClient.SendAsync(firstUploadRequest);

                    var firstUploadResult = await firstUploadResponse.Content.ReadAsStringAsync();

                    Console.WriteLine("First upload response received.");
                    Console.WriteLine(firstUploadResult);

                    JObject firstUploadResultJson = JObject.Parse(firstUploadResult);
                    var firstUploadedID = firstUploadResultJson["files"][0]["id"];

                    using (var secondUploadRequest = new HttpRequestMessage(HttpMethod.Post, "upload"))
                    {
                        secondUploadRequest.Headers.TryAddWithoutValidation("Api-Key", apiKey);
                        secondUploadRequest.Headers.Accept.Add(new("application/json"));

                        var secondUploadByteArray = File.ReadAllBytes(secondPath);
                        var secondUploadByteAryContent = new ByteArrayContent(secondUploadByteArray);
                        secondUploadByteAryContent.Headers.TryAddWithoutValidation("Content-Type", "application/octet-stream");
                        secondUploadByteAryContent.Headers.TryAddWithoutValidation("Content-Filename", Path.GetFileName(secondPath));


                        secondUploadRequest.Content = secondUploadByteAryContent;
                        var secondUploadResponse = await httpClient.SendAsync(secondUploadRequest);

                        var secondUploadResult = await secondUploadResponse.Content.ReadAsStringAsync();

                        Console.WriteLine("Second upload response received.");
                        Console.WriteLine(secondUploadResult);

                        JObject secondUploadResultJson = JObject.Parse(secondUploadResult);
                        var secondUploadedID = secondUploadResultJson["files"][0]["id"];
                        using (var mergeRequest = new HttpRequestMessage(HttpMethod.Post, "merged-pdf"))
                        {
                            mergeRequest.Headers.TryAddWithoutValidation("Api-Key", apiKey);
                            mergeRequest.Headers.Accept.Add(new("application/json"));

                            mergeRequest.Headers.TryAddWithoutValidation("Content-Type", "application/json");


                            JObject parameterJson = new JObject
                            {
                                ["id"] = new JArray(firstUploadedID, secondUploadedID),
                                ["pages"] = new JArray(1, 1),
                                ["type"] = new JArray("id", "id"),

                            };

                            mergeRequest.Content = new StringContent(parameterJson.ToString(), Encoding.UTF8, "application/json"); ;
                            var mergeResponse = await httpClient.SendAsync(mergeRequest);

                            var mergeResult = await mergeResponse.Content.ReadAsStringAsync();

                            Console.WriteLine("Processing response received.");
                            Console.WriteLine(mergeResult);
                        }
                    }
                }
            }
        }
    }
}
